### PR TITLE
gtk4: add sample/examples/application2

### PR DESCRIPTION
**Pull request by ToshioCP**
The gtk4/sample/examples/application2 directory is based on the C programs in the following directory in the GTK 4 repository.

- https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application2

The Ruby program doesn't use GResource, which is the biggest difference from the original one.
If you want know the details, see README.md.

### CODE REVIEW

**Comment by kou** [comment]
Let's use inlined UI definitions instead of `.gresource`.

**Comment by ToshioCP** [comment]
A Ruby program in the previous message replaces exampleapp.rb.
In addition, README.md is modified.
Now it describes the difference from the original program with regard to GResouce.

File: gtk4/sample/examples/application2/exampleapp.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# Example from:
+# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application2/exampleapp.c
+# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application2/exampleappwin.c
```
Comment (kou) [inline gtk4/sample/examples/application2/exampleapp.rb:21]:
```suggestion
# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application2/exampleappwin.c
# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application2/window.ui
```

File: gtk4/sample/examples/application2/exampleapp.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# License: LGPL2.1-or-later
+
+require "gtk4"
+
+class ExampleAppWindow < Gtk::ApplicationWindow
+  type_register
+  class << self
+    def init
+      ui_string = <<~EOS
```
Comment (kou) [inline gtk4/sample/examples/application2/exampleapp.rb:30]: How about using more meaningful name?

* Use `template` for variable name
* Use `TEMPLATE` for here document identifier

```suggestion
      template = <<~TEMPLATE
```

File: gtk4/sample/examples/application2/exampleapp.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+              <property name="orientation">vertical</property>
+              <child>
+                <object class="GtkStack" id="stack"/>
+              </child>
+            </object>
+          </child>
+        </template>
+      </interface>
+      EOS
+      set_template(:data => ui_string)
```
Comment (kou) [inline gtk4/sample/examples/application2/exampleapp.rb:48]: Could you use `XXX: YYY` style?

```suggestion
      set_template(data: ui_string)
```

File: gtk4/sample/examples/application2/exampleapp.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    end
+  end
+
+  def initialize(application)
+    super(:application => application)
```
Comment (kou) [inline gtk4/sample/examples/application2/exampleapp.rb:53]:
```suggestion
    super(application: application)
```

**Comment by ToshioCP** [comment]
I made changes for exampleapp.rb along your suggestion.

**Comment by kou** [comment]
Thanks.

Could you update the pull request description to reflect the current changes before we merge? We use the pull request description as a commit message.

**Comment by ToshioCP** [comment]
The pull request description means the first comment, right?
I've editted it for a commit message.
And I changed the title to fit the previous pull request title.

**Comment by kou** [comment]
Right. Thanks.
I've merged.

**Pull request merged**